scripts/configurator/parser/cmake_var.py

```diff
@@ -1,9 +1,108 @@
-from typing import Dict, Any, Type, Mapping
+from typing import Dict, Any, Optional, Type, Mapping, List
+import re
+from sys import stderr
+
+
+class metadata:
+   keyword_expr = re.compile(r"([A-Z_]+)")
+
+   def __init__(self, comment: List[str]) -> None:
+      self.depends_on: Optional[List[str]] = None
+      self.group: Optional[str] = None
+      self.realtype: Optional[str] = None
+
+      self.kw_to_parse: Mapping[str, function] = {
+         "REALTYPE" : self.parse_realtype,
+         "DEPENDS_ON" : self.parse_depends_on,
+         "GROUP" : self.parse_group
+      }
+
+      self.kw_to_setter = {
+         "REALTYPE": self.get_setter_realtype(),
+         "DEPENDS_ON": self.get_setter_depends_on(),
+         "GROUP": self.get_setter_group(),
+      }
+
+      self.kw_to_var = {
+         "REALTYPE": self.realtype,
+         "DEPENDS_ON": self.depends_on,
+         "GROUP": self.group,
+      }
+
+      for i, line in enumerate(comment):
+         match = self.keyword_expr.match(line)
+         if match:
+            key = match.groups()[0]
+            if i == 0:
+               if key in self.kw_to_parse:
+                  msg = """WARNING: First line uses a reserved keyword: {}.
+                  This will be ignored."""
+                  stderr.write(msg.format(key))
+               continue
+            elif key not in self.kw_to_parse:
+               msg = "{} is not a valid metadata keyword."
+               raise ValueError(msg.format(key))
+            elif self.kw_to_var[key] is not None:
+               msg = "Key {} has already a value associated to itself."
+               raise ValueError(msg.format(key))
+            parse_function = self.kw_to_parse[key]
+            not_matched = line[len(key):].lstrip()
+            setter = self.kw_to_setter[key]
+            s = parse_function(not_matched)
+            setter(s)
+
+   def get_setter_depends_on(self):
```

Comment:
After some thinking, I believe the amount of boilerplate code introduce by this setters is way too high, compared to a trivial if/elseif list. Therefore, we should really use just one generic setter like this:

```python

def get_setter(self, var_name):
   def setter(val):
      nonlocal self
      self.__dict__[var_name] = val
    return setter
```

Reply:
Ok, but what about validation?

Reply:
What validation? That `var_name` really exists? You could do `assert(var_name in self.__dict__)`.
Note that `var_name` is not something that comes from the outer world. It's simply hard-coded in the dictionary:

```
"GROUP": self.get_setter("group")
```

Etc. So it's safe to do. Got it?

Reply:
What i meant was more like input validation. If i remember correctly, the user cannot pass any parameter to our setter and expect it to be correct